src/lints/type_method_marked_deprecated.ron

```diff
@@ -0,0 +1,72 @@
+    id: "type_method_marked_deprecated",
+    human_readable_name: "type method #[deprecated] added",
+    description: "A method in a type implementation has been newly marked with #[deprecated].",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+                        owner_type: __typename @tag @output
+
+                        importable_path {
+                            path @tag @output
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            
+                            method {
+                                method_name: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                deprecated @filter(op: "=", value: ["$true"])
+                            }
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                            end_line @output
+                        }
+                    }
+                }
+            }
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        __typename @filter(op: "=", value: ["%owner_type"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+
+                            method {
+                                name @filter(op: "=", value: ["%method_name"])
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                deprecated @filter(op: "!=", value: ["$true"])
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+    },
+    error_message: "A type method is now #[deprecated]. Downstream crates will get a compiler warning when using this method.",
+    per_result_error_template: Some("method {{method_name}} in {{owner_type}} {{join \"::\" path}} in {{span_filename}}:{{span_begin_line}}"),
```

Comment:
Usually it's best to keep the per-line templates as short and consistent as possible. Concretely:
- Whenever there's readily-available Rust syntax that can make things shorter, let's use it unless it'll be confusing.
- The emphasis here is on the method, and whether it's on a struct/enum/union isn't _super_ important right now. We can omit that.
- Consistency is quite important. Here, `{{owner_type}}` would have come out capitalized (which isn't legal Rust!) so it would have been confusing.
```suggestion
    per_result_error_template: Some("method {{join \"::\" path}}::{{method_name}} in {{span_filename}}:{{span_begin_line}}"),
```